packages/@aws-cdk/aws-ecs/lib/container-definition.ts

```diff
@@ -562,6 +575,51 @@ function renderUlimit(ulimit: Ulimit): CfnTaskDefinition.UlimitProperty {
     hardLimit: ulimit.hardLimit,
   };
 }
+/**
+ * Container dependency
+ */
+export interface ContainerDependency {
+  /**
+   * The container to depend on
+   */
+  readonly container: ContainerDefinition,
+  /**
+   * The dependency condition of the container
+   */
+  readonly condition?: ContainerDependencyCondition,
+}
+
+export enum ContainerDependencyCondition {
+  /**
+   * This condition emulates the behavior of links and volumes today.
+   * It validates that a dependent container is started before permitting other containers to start.
+   */
+  START = 'START',
+
+  /**
+   * This condition validates that a dependent container runs to completion (exits) before permitting other containers to start.
+   * This can be useful for nonessential containers that run a script and then exit.
+   */
+  COMPLETE = 'COMPLETE',
+
+  /**
+   * This condition is the same as COMPLETE, but it also requires that the container exits with a zero status.
+   */
+  SUCCESS = 'SUCCESS',
+
+  /**
+   * This condition validates that the dependent container passes its Docker health check before permitting other containers to start.
+   * This requires that the dependent container has health checks configured. This condition is confirmed only at task startup.
+   */
+  HEALTHY = 'HEALTHY',
+}
+
+function renderContainerDependency(containerDependency: ContainerDependency): CfnTaskDefinition.ContainerDependencyProperty {
+  return {
+    containerName: containerDependency.container.node.id,
+    condition: containerDependency.condition || ContainerDependencyCondition.HEALTHY
```

Comment:
Looks like you're defining Healthy as the default, please document this as the default.